front/src/modules/ui/navbar/components/NavCollapseButton.tsx

```diff
@@ -29,15 +32,27 @@ const StyledCollapseButton = styled.button`
   padding: 0;
 
   user-select: none;
+  visibility: ${({ isHovered }) => (isHovered ? 'visible' : 'hidden')};
   width: 32px;
+
+  @keyframes fadeIn {
+    0% {
+      opacity: 0;
+    }
+    100% {
+      opacity: 1;
+    }
+  }
 `;
 
 type CollapseButtonProps = {
   direction?: 'left' | 'right';
+  isVisible: boolean;
```

Comment:
I'm not too sure about the API where developers need to explicitly state that the component is visible. Can we change this to have the component visible by default, and use a `hide` props instead ?

Reply:
```suggestion
  hide: boolean;
```